src/Classes/Utils/Presets.ts

```diff
@@ -7,22 +7,21 @@ import { ShapeType } from "../../Enums/ShapeType";
 import { HoverMode } from "../../Enums/Modes/HoverMode";
 
 export class Presets {
-    public static getPreset(key: PresetType): RecursivePartial<IOptions> | undefined {
-        return this.available[key.toString()];
+    private static presets: { [preset: string]: RecursivePartial<IOptions> } = {};
+
+    public static getPreset(preset: PresetType | string): RecursivePartial<IOptions> | undefined {
+        return this.presets[preset];
     }
 
-    private static get available(): { [key: string]: RecursivePartial<IOptions> } {
-        return {
-            [PresetType.basic]: this.basic,
-            [PresetType.backgroundMask]: this.backgroundMask,
-            [PresetType.fontAwesome]: this.fontAwesome,
-            [PresetType.snow]: this.snow,
-            [PresetType.bouncing]: this.bouncing,
-            [PresetType.stars]: this.stars,
+    public static addPreset(presetKey: string, options: RecursivePartial<IOptions>) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'addPreset' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5274115)